### PR TITLE
Enhancement: Refactor Visualisation Tab

### LINKED ISSUE #127: Enhancement: Improve Data Visualisation section
### Description
With the introduction of the TLGs tab, the Outputs section now is no longer required for reporting but rather for enhanced data visualisation. We want to include a number of different possibilities for the User to carry out in depth exploratory analysis of the data without needing outputs for reporting.

This issue is to modify the basic setup/layout of the outputs tab, so that it is ready for more detailed changes to be made.

### Definition of Done

- [ ] rename outputs to visualisation tab
- [ ] Modify layout to make for better exploratory analysis- bigger plots

### PR DESCRIPTION
**Pull request by js3110**
## Issue

Closes #127 

## Description

With the introduction of the TLGs tab, the Outputs section now is no longer required for reporting but rather for enhanced data visualisation. We want to change the layout so it is better for visualisation.

### Definition of Done

- [x] rename outputs to visualisation tab
- [x] Modify layout to make for better exploratory analysis- bigger plots

## How to test
Check visualisation tab layout

## Contributor checklist
- [x] Code passes lintr checks
- [x] Code passes all unit tests
- [x] New logic covered by unit tests
- [x] New logic is documented

## Notes to reviewer

The plan is to reformat this section using bslib, which will then need to be implemented across the rest of the app. Refactoring of the Data section is being covered by #140

### CODE REVIEW

File: inst/shiny/modules/tab_visuals.R [hunk 1]
```diff
@@ -0,0 +1,560 @@
...
+      selectInput(ns("analytemean"), "Choose the Analyte:", choices = sort(y),
+                  selected = y[1])
+    })
+
+    # select the study id for the mean plot
+    # (needed for preclinical data, where multiple studyids per dataset)
+    output$studyidmean <- renderUI({
+      y <- data() %>%
+        pull(STUDYID) %>%
+        unique()
```
Comment (m-kolomanski) [inline inst/shiny/modules/tab_visuals.R:238]: suggestion: This pattern is repeated a lot in the app, might be worth putting it into a function in some kind of `utils` file? It is debatable if it is worth it, so whatever you feel more comfortable with.

```R
pull_unique <- function(data, colname) {
  data |> pull(colname) |> unique()
}
```

File: inst/shiny/modules/tab_visuals.R [hunk 2]
```diff
@@ -0,0 +1,560 @@
...
+      )
+    })
+
+    # Save summary stats to improve
+    observeEvent(input$downloadsum, {
+      showModal(modalDialog(
+        title = "Please enter the path to the folder on improve for your results:",
+        textInput(ns("pathresults"), "Path:"),
+        textInput(ns("filename"), "Enter filename of choice:", ""),
+        actionButton(ns("go2"), "GO"),
```
Comment (m-kolomanski) [inline inst/shiny/modules/tab_visuals.R:380]: issue: This button does not seem to have any event attached.
issue: If this is functional button, it needs more descriptive name.,

File: inst/shiny/modules/tab_visuals.R [hunk 3]
```diff
@@ -0,0 +1,560 @@
...
+            startsWith(PPTESTCD, "aucint"),
+            paste0(PPTESTCD, "_", start, "-", end),
+            PPTESTCD
+          )
+        )
+    })
+
+    # select which parameter to box or violin plot
+    output$selectboxplot <- renderUI({
+      param_choices <- boxplotdata()$PPTESTCD %>% unique()
```
Comment (m-kolomanski) [inline inst/shiny/modules/tab_visuals.R:423]: issue: Please avoid mixing base R and dplyr patterns.
```suggestion
      param_choices <- boxplotdata() %>%
        pull(PPTESTCD) %>%
        unique()
```

File: inst/shiny/modules/tab_visuals.R [hunk 4]
```diff
@@ -0,0 +1,560 @@
...
+                      input$selected_colorvars_boxplot), {
+                   xvar_options_list <- lapply(
+                     c(input$selected_xvars_boxplot, input$selected_colorvars_boxplot),
+                     \(id_var) paste(id_var, unique(boxplotdata()[[id_var]]), sep = ": ")
+                   )
+
+                   names(xvar_options_list) <- c(
+                     input$selected_xvars_boxplot,
+                     input$selected_colorvars_boxplot
+                   )
```
Comment (m-kolomanski) [inline inst/shiny/modules/tab_visuals.R:464]: suggestion: You can use `setNames` to make this a tiny bit nicer.
```suggestion
                   xvar_options_list <- lapply(
                     c(input$selected_xvars_boxplot, input$selected_colorvars_boxplot),
                     \(id_var) paste(id_var, unique(boxplotdata()[[id_var]]), sep = ": ")
                   ) |>
                     setNames(c(input$selected_xvars_boxplot, input$selected_colorvars_boxplot))
```

File: inst/shiny/modules/tab_visuals.R [hunk 5]
```diff
@@ -0,0 +1,560 @@
...
+    concat_report_data <- reactive({
+      data <- res_nca()
+      data$formatted_res <- boxplotdata()
+      data
+    })
+
+    # render the markdown upon clicking generate
+    rendered_rmd <- reactiveVal(NULL)
+
+    observeEvent(input$generate_report, {
```
Comment (m-kolomanski) [inline inst/shiny/modules/tab_visuals.R:523]: issue: Seems to be related to the commented out code. As of now it is dead code, should be removed / moved alongside the commented out parts.

**Review by m-kolomanski — CHANGES_REQUESTED** [review]
Hey, good job in general, must have taken a lot of effort to get that together! The UI looks quite nice, my only suggestion would be to maybe try and work a bit with the widths of different elements (the drag-and-drop, buttons etc) to make it a bit more tight.

I also have some points of discussion, mainly in terms of code style and format.

File: inst/shiny/modules/tab_visuals.R (hunk 1 above)
Comment (js3110) [reply inst/shiny/modules/tab_visuals.R:238]: Personally I think its not worth it for now, but if we end up using it even more later then we can implement it then

File: inst/shiny/modules/tab_visuals.R [hunk 6]
```diff
@@ -0,0 +1,560 @@
...
+          scrollY = TRUE,
+          lengthMenu = list(c(10, 25, -1), c("10", "25", "All")),
+          fixedHeader = TRUE
+        )
+      )
+    })
+
+    output$descriptivestats2 <- DT::renderDataTable({
+      req(res_nca())
+      DT::datatable(
```
[the remark this answers is not in the document]
Comment (js3110) [reply inst/shiny/modules/tab_visuals.R:362]: yes i do

File: inst/shiny/modules/tab_visuals.R [hunk 7]
```diff
@@ -0,0 +1,560 @@
...
+                          pickerInput(
+                            inputId = ns("selected_varvalstofilter_boxplot"),
+                            label = "Select values to display",
+                            multiple = TRUE,
+                            choices = NULL,
+                            selected = NULL,
+                            options = list(`actions-box` = TRUE)
+                          ),
+                          uiOutput(ns("violin_toggle")),
+                          position = "right",
```
[the remark this answers is not in the document]
Comment (js3110) [reply inst/shiny/modules/tab_visuals.R]: @m-kolomanski I have tried to do as you've shown in the commit linked, and have just followed the indentation that Rstudio automatically creates, however I am still getting flagged with lintr. I have pushed the changes (along with making the picker inputs static)- can you take a look and figure out why lintr still hates me please .

File: inst/shiny/modules/tab_visuals.R [hunk 8]
```diff
@@ -0,0 +1,533 @@
...
+          )
+        ),
+        plotlyOutput(ns("individualplot"))
+      )
+    ),
+    nav_panel("Mean Plots",
+      layout_sidebar(
+        sidebar = sidebar(
+          selectInput(
+            inputId = ns("analytemean"),
```
Comment (m-kolomanski) [inline inst/shiny/modules/tab_visuals.R:62]: nitpick: Would be nice to clean up IDs to be more readable and break up this kind of pattern:

`analytemean` -> `analyte_mean`
`selectidvar` -> `select_id_var`

and maybe be more verbose with naming some of the ids:

`paramselect` -> `select_display_parameters`

It is kinda hard to read IDs like `selected_varvalstofilter_boxplot`.

File: inst/shiny/modules/tab_visuals.R [hunk 9]
```diff
@@ -0,0 +1,533 @@
...
+        textInput(ns("filename"), "Enter filename of choice:", ""),
+        actionButton(ns("go2"), "GO"),
+        footer = modalButton("Close")
+      ))
+    })
+
+    # alternatively: save locally
+    output$download_browser <- downloadHandler(
+      filename = function() {
+        paste("NCA_summary.csv", sep = "_")
```
Comment (m-kolomanski) [inline inst/shiny/modules/tab_visuals.R:430]: nitpick: `sep = "_"` is redundant in this case.

```suggestion
        paste("NCA_summary.csv")
```

unless something else than constant string was supposed to go into the name?

**Review by m-kolomanski — APPROVED** [review]
I have approved the PR, code looks good, great work! I have left some minor nitpicks, but I consider them non-blocking.

File: inst/shiny/modules/tab_visuals.R (hunk 7 above)
[the remark this answers is not in the document]
Comment (m-kolomanski) [reply inst/shiny/modules/tab_visuals.R]: As you have noticed, I have pushed changes that mostly fix the issue. When not maintained from the very start, the indentation can indeed be quite hard to unwrap and the behavior of Rstudio formatter does not help in those situations.

**Review by Gero1999 — APPROVED** [review]
lgtm

**Pull request merged**